src/Graph.ts

```diff
@@ -51,11 +52,11 @@ export default class Graph {
 	context: string;
 	externalModules: ExternalModule[] = [];
 	getModuleContext: (id: string) => string;
-	hasLoaders: boolean;
 	isPureExternalModule: (id: string) => boolean;
 	moduleById = new Map<string, Module | ExternalModule>();
 	assetsById = new Map<string, Asset>();
 	modules: Module[] = [];
+	needsTreeshakingPass: boolean = false;
```

Comment:
Could you perhaps explain what this refactoring fixed in the process if there was an underlying fix with this?

Reply:
Actually there was though I no longer know the precise situation. Before, each module had its own `needsTreeshakingPass`. Now if a variable is declared in module 'a'  but included by module 'b', it could happen that the flag would be set for module 'a' instead of module 'b'. If this module was already processed but no other modules request a new pass, it could happen that no new pass was triggered. Having a shared flag for all modules makes much more sense and fixes this kind of issue.
BTW as part of the refactorings of the other PR, I could get rid of `requestTreeshakingPass` on the AST context entirely and keep this logic to the module level.